executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -222,24 +212,40 @@ impl<'a, T: FieldElement> MachineExtractor<'a, T> {
             .filter_map(|(i, &pf)| (!extracted_prover_functions.contains(&i)).then_some(pf))
             .collect::<Vec<_>>();
 
+        // In the remaining witness, we might still have some multiplicity columns
```

Comment:
Maybe relevant here: the way we currently "call" the main machine is by constraining the operation id on the first row. This is done by the linker. While refactoring it in #2359, I thought about instead sending to the bus. I did not make this change though.

Reply:
I don't quite see the connection to this PR, but interesting idea!

That's my understanding of what [Ceno](https://georgwiese.github.io/crypto-summaries/Papers/Ceno---Non-uniform,-Segment-and-Parallel-Zero-knowledge-Virtual-Machine) does: There is no main machine, each instruction calls the next one, but the very first instruction get's called by the verifier doing a bus send (basically initializing the bus accumulator with a non-zero value).

But how exactly would it work here? The main machine would have a run operation (with no arguments)? Who initiates the call, is it also the verifier? One problem I'd see with this is that then you can't express it with lookups / permutations anymore, so we'd have to drop support for that entirely.